lib/promoted/ruby/client/request_builder.rb

```diff
@@ -2,52 +2,49 @@ module Promoted
   module Ruby
     module Client
       class RequestBuilder
-        attr_reader   :session_id, :user_info,
+        attr_reader   :session_id, :only_log, :experiment,
                       :view_id, :insertion, :to_compact_delivery_insertion_func,
                       :request_id, :full_insertion, :use_case, :request, :to_compact_metrics_insertion_func
 
-        def initialize params={}
-          @only_log                = params[:only_log] || false
-        end
+        attr_accessor :timing, :user_info, :platform_id
+
+        def initialize;end
 
         # Populates request parameters from the given arguments, presumed to be a hash of symbols.
         def set_request_params args = {}
           @request                 = args[:request] || {}
+          @experiment              = args[:experiment]
```

Comment:
side-note: We can chat about this naming the next we're in person.

Reply:
Sometimes it's sort of "experiment", sometimes, it's sort of "cohort_membership".